docs/codegen/src/rules_docs.rs

```diff
@@ -0,0 +1,464 @@
+use pglt_analyse::{AnalyserOptions, AnalysisFilter, RuleFilter, RuleMetadata};
+use pglt_analyser::{Analyser, AnalyserConfig};
+use pglt_console::StdDisplay;
+use pglt_diagnostics::{Diagnostic, DiagnosticExt, PrintDiagnostic};
+use pglt_query_ext::diagnostics::SyntaxDiagnostic;
+use pglt_workspace::settings::Settings;
+use pulldown_cmark::{CodeBlockKind, Event, LinkType, Parser, Tag, TagEnd};
+use std::{
+    fmt::Write as _,
+    fs,
+    io::{self, Write as _},
+    path::Path,
+    slice,
+    str::{self, FromStr},
+};
+
+/// Generates the documentation page for each lint rule.
+///
+/// * `docs_dir`: Path to the docs directory.
+pub fn generate_rules_docs(docs_dir: &Path) -> anyhow::Result<()> {
+    let rules_dir = docs_dir.join("rules");
+
+    if rules_dir.exists() {
+        fs::remove_dir_all(&rules_dir)?;
+    }
+    fs::create_dir_all(&rules_dir)?;
+
+    let mut visitor = crate::utils::LintRulesVisitor::default();
+    pglt_analyser::visit_registry(&mut visitor);
+
+    let crate::utils::LintRulesVisitor { groups } = visitor;
+
+    for (group, rules) in groups {
+        for (rule, metadata) in rules {
+            let content = generate_rule_doc(group, rule, metadata)?;
+            let dashed_rule = Case::Kebab.convert(rule);
+            fs::write(rules_dir.join(format!("{}.md", dashed_rule)), content)?;
+        }
+    }
+
+    Ok(())
+}
+
+fn generate_rule_doc(
+    group: &'static str,
+    rule: &'static str,
+    meta: RuleMetadata,
+) -> Result<String> {
+    let mut content = Vec::new();
+
+    writeln!(content, "# {rule}")?;
+
+    writeln!(
+        content,
+        "**Diagnostic Category: `lint/{}/{}`**",
+        group, rule
+    )?;
+
+    let is_recommended = meta.recommended;
+
+    // add deprecation notice
+    if let Some(reason) = &meta.deprecated {
+        writeln!(content, "> [!WARNING]")?;
+        writeln!(content, "> This rule is deprecated and will be removed in the next major release.\n**Reason**: {reason}")?;
+    }
+
+    writeln!(content)?;
+    writeln!(content, "**Since**: `v{}`", meta.version)?;
+    writeln!(content)?;
+
+    // add recommended notice
+    if is_recommended {
+        writeln!(content, "> [!NOTE]")?;
+        writeln!(
+            content,
+            "> This rule is recommended. A diagnostic error will appear when linting your code."
+        )?;
+    }
+
+    writeln!(content)?;
+
+    // add source information
+    if !meta.sources.is_empty() {
+        writeln!(content, "**Sources**: ")?;
+
+        for source in meta.sources {
+            let rule_name = source.to_namespaced_rule_name();
+            let source_rule_url = source.to_rule_url();
+            write!(content, "- Inspired from: ")?;
+            writeln!(
+                content,
+                "<a href=\"{source_rule_url}\" target=\"_blank\"><code>{rule_name}</code></a>"
+            )?;
+        }
+        writeln!(content)?;
+    }
+
+    write_documentation(group, rule, meta.docs, &mut content)?;
```

Comment:
I just realized that the `declare_lint_rule` macro will use doc comment and attach it as a `.doc` property. I'm impressed 🤯